### PR TITLE
fix(post): update logic to reject bodies for GET/HEAD 

### PR DESCRIPTION
**Pull request by vceder**
But allow it for other types

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ function createRemixRequest(request, options = {}) {
     headers: request.headers,
     signal: controller.signal,
     // eslint-disable-next-line unicorn/no-null -- Request init expects a `null` value.
-    body: isGetOrHead(request) ? request.body : null,
+    body: isGetOrHead(request) ? null : request.clone().body,
   };
```
Comment (mettson) [inline src/server.js:74]: Do we want to leave the `.clone()` here?

**Review by mettson — DISMISSED** [review]
Looks good, just left a question!

File: src/server.js (hunk 1 above)
Comment (lfantone) [reply src/server.js:74]: You don't really need a `clone` in this context as a new request is being created below.

Comment (lfantone) [reply src/server.js:75]: Wouldn't that be an issue within the middleware provider?

File: src/server.js [hunk 2]
```diff
@@ -70,7 +70,8 @@ function createRemixRequest(request, options = {}) {
     headers: request.headers,
     signal: controller.signal,
     // eslint-disable-next-line unicorn/no-null -- Request init expects a `null` value.
-    body: isGetOrHead(request) ? request.body : null,
+    body: isGetOrHead(request) ? null : request.clone().body,
+    duplex: request.method !== 'GET' && request.method !== 'HEAD' ? 'half' : undefined,
```
Comment (lfantone) [inline src/server.js:74]: Wouldn't the suggestion be the same?

```suggestion
    duplex: isGetOrHead(request) ? undefined : 'half',
```

**Review by lfantone — APPROVED** [review]

**Review by rommysas — APPROVED** [review]
Everything seems good to me

**Pull request merged**